addons/modules/functions/fnc_moduleCrewToGunner.sqf

```diff
@@ -0,0 +1,49 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Zeus module function to have AI crew take the place of an incapacitated gunner.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC] call zen_modules_fnc_moduleCrewToGunner
+ *
+ * Public: No
+ */
+
+params ["_logic"];
+
+private _vehicle = attachedTo _logic;
+deleteVehicle _logic;
+
+if (isNull _vehicle) exitWith {
+    [LSTRING(NoObjectSelected)] call EFUNC(common,showMessage);
+};
+
+if (!alive _vehicle) exitWith {
+    [LSTRING(OnlyAlive)] call EFUNC(common,showMessage);
+};
+
+if (fullCrew [_vehicle, "gunner", true] isEqualTo []) exitWith {
+    [LSTRING(MustHaveGunner)] call EFUNC(common,showMessage);
+};
+
+_vehicle allowCrewInImmobile true;
+
+private _gunner = gunner _vehicle;
+private _newGunners = (crew _vehicle - [_gunner]) select {canMove _x};
+
+if (!isNull _gunner) then {
+    [QGVAR(teleportOutOfVehicle), [_gunner], _gunner] call CBA_fnc_targetEvent;
```

Comment:
If the only thing you are doing with the event is moving the unit out. You can just use the `moveOut` command directly; it is AG EG.

Reply:
Wow! I supposed I've never actually looked at that for moveOut and just assumed from moveIn____

Reply:
https://github.com/zen-mod/ZEN/pull/423/commits/f22dfedc7d46e9069613ddad889138f39e6b45a8